supertokens_python/recipe/session/session_request_functions.py

```diff
@@ -391,9 +405,28 @@ async def refresh_session_in_request(
                 )
             )
 
-        log_debug_message(
-            "refreshSession: UNAUTHORISED because refresh_token in request is None"
-        )
+        # We need to clear the access token cookie if
+        # - the refresh token is not found, and
+        # - the allowed_transfer_method is 'cookie' or 'any', and
+        # - an access token cookie exists (otherwise it'd be a no-op)
+        # See: https://github.com/supertokens/supertokens-node/issues/790
+        if (
+            allowed_transfer_method in ("cookie", "any")
+            and get_token(request, "access", "cookie") is not None
+        ):
+            response_mutators.append(
+                set_cookie_response_mutator(
+                    config,
+                    get_cookie_name_from_token_type("access"),
+                    "",
+                    0,
+                    "access_token_path",
+                    request,
+                )
+            )
+            log_debug_message(
+                "refreshSession: cleared access token and returning UNAUTHORISED because refresh_token in request is None"
+            )
```

Comment:
it's a little strange that here we are clearing tokens, but, the error that's thrown below has `clear_tokens=False,`. Something can be refactored / changed to make things more clear? (This comment also applies to node and golang PR).

Also in node (writing it here since node pr is merged), we are clearing the access token cookie from the `res` object that's an input to the `refreshSessionInRequest` function, but right after, we are throwing an UNAUTHORISED error. Now the errorHandler will catch that, but are we always sure that the `res` object on which that error handler operates on is the same `res` object on which we cleared the access token? You will have to check how the various frameworks in node implement the middleware to catch errors (is the `res` object given to the errorHandler always the same instance as the one given to the middleware (from supertokens.ts file)?

Reply:
1. we decided to use `clearTokens: True`.

2. Took a look at all the frameworks for node. We indeed use the same request object.